src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -76,6 +54,65 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+
+        internal async Task<HttpResponseMessage> CreateCheckStatusResponse(
+            HttpRequestMessage request,
+            string instanceId,
+            OrchestrationClientAttribute attribute,
+            int totalTimeout,
+            int retryTimeout)
+        {
+            GetClientResponseLinks(request, instanceId, attribute, out var statusQueryGetUri, out var sendEventPostUri,
+                out var terminatePostUri);
+
+            if (totalTimeout < retryTimeout)
+            {
+                throw new InvalidOperationException($"Total timeout {totalTimeout} should be bigger than {retryTimeout}");
+            }
+
+            var iterationCount = totalTimeout / retryTimeout;
+            JToken durableFunctionOutput = null;
+
+            var client = GetClient(request);
+            var status = await client.GetStatusAsync(instanceId);
+            for (var i = 0; i < iterationCount; i++)
+            {
+                Thread.Sleep(retryTimeout);
+                if (status != null && status.RuntimeStatus == OrchestrationRuntimeStatus.Completed)
```

Comment:
We need to also handle the case where the orchestration status is in other terminal states, like `Failed`, `Terminated`, or `Canceled`. In those cases, it probably makes sense to return an error status value.